lib/core/controlplane.rb

```diff
@@ -143,40 +143,38 @@ def query_workloads(workload, a_gvc = gvc, a_org = org, partial_workload_match:
     api.query_workloads(org: a_org, gvc: a_gvc, workload: workload, gvc_op_type: gvc_op, workload_op_type: workload_op)
   end
 
-  def workload_get_replicas(workload, location:)
-    cmd = "cpln workload get-replicas #{workload} #{gvc_org} --location #{location} -o yaml"
+  def fetch_workload_replicas(workload, location:)
+    cmd = "cpln workload replica get #{workload} #{gvc_org} --location #{location} -o yaml"
     perform_yaml(cmd)
   end
 
-  def workload_get_replicas_safely(workload, location:)
-    cmd = "cpln workload get-replicas #{workload} #{gvc_org} --location #{location} -o yaml"
-
-    Shell.debug("CMD", cmd)
-
-    result = Shell.cmd(cmd, capture_stderr: true)
-    YAML.safe_load(result[:output]) if result[:success]
+  def stop_workload_replica(workload, replica, location:)
+    cmd = "cpln workload replica stop #{workload} #{gvc_org} --replica-name #{replica} --location #{location}"
+    perform(cmd, output_mode: :none)
   end
 
   def fetch_workload_deployments(workload)
     api.workload_deployments(workload: workload, gvc: gvc, org: org)
   end
 
-  def workload_deployment_version_ready?(version, next_version, expected_status:)
+  def workload_deployment_version_ready?(version, next_version)
     return false unless version["workload"] == next_version
 
     version["containers"]&.all? do |_, container|
-      ready = container.dig("resources", "replicas") == container.dig("resources", "replicasReady")
-      expected_status == true ? ready : !ready
+      container.dig("resources", "replicas") == container.dig("resources", "replicasReady")
     end
   end
 
-  def workload_deployments_ready?(workload, expected_status:)
+  def workload_deployments_ready?(workload, location:, expected_status:)
+    deployed_replicas = fetch_workload_replicas(workload, location: location)["items"].length
+    return deployed_replicas.zero? if expected_status == false
```

Comment:
With the previous check, sometimes it would return `true` even though there were still replicas deployed when waiting for the workload to not be ready. Now we check explicitly that the number of deployed replicas is zero, which is more accurate.